src/Fixer/FunctionNotation/NativeFunctionInvocationFixer.php

```diff
@@ -0,0 +1,191 @@
+                    && 0 === $openBracesCount
+                ) {
+                    $isWithinNamespace = false;
+                }
+
+                continue;
+            }
+
+            if ($token->isGivenKind(T_NAMESPACE)) {
+                $namespaceDeclarationEndIndex = $tokens->getNextTokenOfKind($index, array(
+                    ';',
+                    '{',
+                ));
+
+                if ($tokens[$namespaceDeclarationEndIndex]->getContent() === '{') {
+                    $isNamespaceWithBraces = true;
+                }
+
+                $isWithinNamespace = true;
+            }
+
+            if (!$isWithinNamespace) {
+                continue;
+            }
+
+            // test if we are at a function call
+            if (!$token->isGivenKind(T_STRING)) {
+                continue;
+            }
+
+            $next = $tokens->getNextMeaningfulToken($index);
+            if (!$tokens[$next]->equals('(')) {
+                continue;
+            }
+
+            $functionNamePrefix = $tokens->getPrevMeaningfulToken($index);
+            if ($tokens[$functionNamePrefix]->isGivenKind(array(T_DOUBLE_COLON, T_NEW, T_OBJECT_OPERATOR, T_FUNCTION))) {
+                continue;
+            }
+
+            if ($tokens[$functionNamePrefix]->isGivenKind(T_NS_SEPARATOR)) {
+                // skip if the call is to a constructor or to a function in a namespace other than the default
+                $prev = $tokens->getPrevMeaningfulToken($functionNamePrefix);
+                if ($tokens[$prev]->isGivenKind(array(T_STRING, T_NEW))) {
+                    continue;
+                }
+            }
+
+            if (!in_array(strtolower($tokenContent), $internalFunctionNames, true)) {
+                continue;
+            }
+
+            // do not bother if previous token is already namespace separator
+            if ($tokens[$index - 1]->isGivenKind(T_NS_SEPARATOR)) {
+                continue;
+            }
+
+            $indexes[] = $index;
+        }
+
+        $indexes = array_reverse($indexes);
+
+        $namespaceSeparator = new Token(array(
+            T_NS_SEPARATOR,
+            '\\',
+        ));
+
+        foreach ($indexes as $index) {
+            $tokens->insertAt($index, $namespaceSeparator);
+        }
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getDefinition()
+    {
+        return new FixerDefinition(
+            'Add leading `\` before function invocation of internal function within namespaces to speed up resolving.',
+            array(new CodeSample(
+'<?php
+
+namespace Foo;
+
+class Bar
+{
+    public function __construct($options) 
+    {
+        if (!array_key_exists("foo", $options)) {
+            throw new \InvalidArgumentException();
+        }
+    }
+    
+}'
+            )),
+            null,
+            null,
+            null,
+            'Risky if a function with the same name as a native function exists in the current namespace'
```

Comment:
While I find this fixer really useful, it's not only case when it't risky. And I would say that the 2nd case is more tricky.

Let me use `is_iterable` function:
```php
var_dump(is_iterable($foo));
```

Then run the fixer on 7.1 env:
```php
var_dump(\is_iterable($foo));
```

Then run the code on 5.6 env - it will fail, because of missing function.
And if I would run original code - it will succeed, as missing function is polyfilled via `symfony/polyfill`.

Please describe a risk about defined function in given env and polyfilled functions.
Also, please add a configuration that I do not want to be prefixed with `\`, so I could set `is_iterable` as exception and still use the rule